build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -425,6 +425,14 @@ message Action {
   bool do_not_cache = 7;
 
   reserved 8; // Used for field moved to [Command][build.bazel.remote.execution.v2.Command].
+
+  // An optional additional salt value used to place this `Action` into a
+  // separate cache namespace from other instances having the same field
+  // contents. This salt typically comes from operational configuration
+  // specific to sources such as repo and service configuration,
+  // and allows disowning an entire set of ActionResults that might have been
+  // poisoned by buggy software or tool failures.
```

Comment:
Why wasn't the tool part of the Action input to begin with?  Wouldn't revving the tool then render a new Action?  Is this pouring salt into the wound of underspecified environments?

Reply:
Action contains a ref to Command so the specific tool paths and environment cause different universes. However internally in our systems if you run something critical like a compiler a million times you sometimes see a zero process exit code but a tool generates no outputs or truncated/torn outputs, causing cache poisoning. Same tool that worked 999,999 times elsewhere but a build failure on missing outputs nonetheless.

Reply:
@erikma isn't ExecuteRequest.skip_cache_lookup meant to address that use case?  Or is this just a case of using the blunt instrument of declaring bankruptcy on one cache namespace as a whole and moving to a clean slate, because the surgical approach may be too expensive?

Reply:
skip_cache_lookup from my POV is a feature flag allowing testing pure remoting without caching overhead. Or possibly for use when the tools involved are generating deliberately nondeterministic outputs, think of an FPGA layout tool or TensorFlow training job doing randomized stochastic gradient descent where you deliberately want to rerun the job without caching.

For the salt it's very much the latter. Imagine if you will 40,000 machines needing the surgical option on one ActionResult entry, or in some cases an entire tree of outputs derived from one bad entry. Better in almost all cases to use the Big Hammer and get to bed earlier.